src/main/java/net/dv8tion/jda/core/entities/MessageActivity.java

```diff
@@ -0,0 +1,246 @@
+     * {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType ActivityType} of the
+     * {@link net.dv8tion.jda.core.entities.MessageActivity MessageActivity} not equals
+     * {@link net.dv8tion.jda.core.entities.MessageActivity.ActivityType#LISTENING ActivityType.LISTENING}.
+     *
+     * <br>Many applications can be found at:
+     * <a href="https://discordapp.com/api/v7/games" target="_blank">https://discordapp.com/api/v7/games</a>.
+     */
+    public static class Application
+    {
+        private final String name;
+        private final String description;
+        private final String iconId;
+        private final String coverId;
+        private final long id;
+
+        public Application(String name, String description, String iconId, String coverId, long id)
+        {
+            this.name = name;
+            this.description = description;
+            this.iconId = iconId;
+            this.coverId = coverId;
+            this.id = id;
+        }
+
+        /**
+         * The name of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications name.
+         */
+        public String getName()
+        {
+            return name;
+        }
+
+        /**
+         * A short description of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications description.
+         */
+        public String getDescription()
+        {
+            return description;
+        }
+
+        /**
+         * The icon id of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications icon id.
+         */
+        public String getIconId()
+        {
+            return iconId;
+        }
+
+        /**
+         * The url of the icon image for this application.
+         *
+         * @return the url of the icon
+         */
+        public String getIconUrl()
+        {
+            return "https://cdn.discordapp.com/application/" + id + "/" + iconId + ".png";
+        }
+
+        /**
+         * The cover aka splash id of this {@link net.dv8tion.jda.core.entities.MessageActivity.Application}.
+         *
+         * @return the applications cover image/id.
+         */
+        public String getCoverId()
+        {
+            return coverId;
+        }
+
+        /**
+         * The url of the cover image for this application.
+         *
+         * @return the url of the cover/splash
+         */
+        public String getCoverUrl()
+        {
+            return "https://cdn.discordapp.com/application/" + id + "/" + coverId + ".png";
+        }
+
+        /**
+         * The id of the current {@link net.dv8tion.jda.core.entities.MessageActivity.Application} as a {@link String}.
+         *
+         * @return the applications id.
+         */
+        public String getId()
+        {
+            return Long.toUnsignedString(id);
```

Comment:
If this is a snowflake this should implement ISnowflake, in which case this method is implemented by the interface